File: internal/models/student.go
package models

import "time"

// Student represents a student in the church service database.
type Student struct {
	ID                  string    `json:"id" gorm:"primaryKey;column:id"`
	FullName            string    `json:"fullName" gorm:"column:full_name"`
	NationalID          string    `json:"nationalId" gorm:"column:national_id"`
	Gender              string    `json:"gender" gorm:"column:gender"`
	BirthDate           string    `json:"birthDate" gorm:"column:birth_date"`
	Governorate         string    `json:"governorate" gorm:"column:governorate"`
	Phone               string    `json:"phone" gorm:"column:phone"`
	ParentPhone         string    `json:"parentPhone" gorm:"column:parent_phone"`
	Address             string    `json:"address" gorm:"column:address"`
	Stage               string    `json:"stage" gorm:"column:stage"`
	Grade               string    `json:"grade" gorm:"column:grade"`
	Track               string    `json:"track" gorm:"column:track"`
	UniversityName      string    `json:"universityName" gorm:"column:university_name"`
	Faculty             string    `json:"faculty" gorm:"column:faculty"`
	StudyYears          string    `json:"studyYears" gorm:"column:study_years"`
	UniversityYear      string    `json:"universityYear" gorm:"column:university_year"`
	CathedralStudentID  string    `json:"cathedralStudentId" gorm:"column:cathedral_student_id"`
	CathedralFamilyID   string    `json:"cathedralFamilyId" gorm:"column:cathedral_family_id"`
	AlexandriaStudentID string    `json:"alexandriaStudentId" gorm:"column:alexandria_student_id"`
	AlexandriaFamilyID  string    `json:"alexandriaFamilyId" gorm:"column:alexandria_family_id"`
	PhotoPath           string    `json:"photoPath" gorm:"column:photo_path"`
	DeaconStatus        bool      `json:"deaconStatus" gorm:"column:deacon_status"`
	Notes               string    `json:"notes" gorm:"column:notes"`
	CreatedAt           time.Time `json:"createdAt" gorm:"column:created_at"`
	UpdatedAt           time.Time `json:"updatedAt" gorm:"column:updated_at"`
}

func (Student) TableName() string { return "students" }

// ImportBatchResult describes the committed result of an Excel import.
type ImportBatchResult struct {
	Inserted int `json:"inserted"`
	Updated  int `json:"updated"`
}

// StageCount represents count per church educational stage.
type StageCount struct {
	Stage string `json:"stage"`
	Count int    `json:"count"`
}

// ChurchSettings stores key-value system configuration.
type ChurchSettings struct {
	Key   string `json:"key" gorm:"primaryKey"`
	Value string `json:"value"`
}

func (ChurchSettings) TableName() string { return "church_settings" }

// NIDData represents parsed details from Egyptian National ID.
type NIDData struct {
	NationalID       string `json:"nationalId"`
	Valid            bool   `json:"valid"`
	BirthDate        string `json:"birthDate"`
	Gender           string `json:"gender"`
	Governorate      string `json:"governorate"`
	Age              int    `json:"age"`
	Error            string `json:"error,omitempty"`
	ChecksumValid    bool   `json:"checksumValid"`
	ExpectedChecksum int    `json:"expectedChecksum"`
	AgeMismatch      bool   `json:"ageMismatch"`
	StageWarning     string `json:"stageWarning,omitempty"`
	SuggestedID      string `json:"suggestedId,omitempty"`
}

File: frontend/src/types/student.ts
export interface Student {
  id: string;
  fullName: string;            // أسم الطالب الرباعي (اجباري)
  familyHead?: string;         // اسم رب الأسرة (اجباري)
  nationalId: string;          // الرقم القومي 14 رقماً (اجباري)
  gender: string;              // النوع (مستخرج آلياً من الرقم القومي)
  birthDate: string;           // تاريخ الميلاد (مستخرج آلياً من الرقم القومي)
  governorate: string;         // المحافظة (مستخرج آلياً من الرقم القومي)
  stage: string;               // المرحلة الحالية (اجباري)
  grade: string;               // الصف الدراسي الحالي (اجباري)
  schoolName?: string;         // اسم المدرسة (اجباري للتعليم المدرسي)
  track?: string;              // نوع / مسار الثانوية (اجباري للثانوي)

  // University Fields (Conditional for 'جامعة' - اجباري لمرحلة الجامعة)
  universityName?: string;     // الجامعة / المعهد (اجباري للجامعة)
  faculty?: string;            // الكلية / التخصص (اجباري للجامعة)
  studyYears?: string;         // عدد سنين الدراسة (اختياري)
  universityYear?: string;     // الفرقة الدراسية / الحالة (اجباري للجامعة)

  // Communication & Address
  parentPhone: string;         // هاتف ولي الأمر (اجباري)
  phone: string;               // رقم التليفون (اختياري)
  address: string;             // العنوان (اختياري)

  // Church & Care IDs
  churchFamilyId?: string;     // رقم الأسرة بكشوفات الكنيسة (اجباري)
  cathedralFamilyId: string;   // رقم الأسرة في برنامج الرعاية الكنسية (اجباري)
  cathedralStudentId?: string; // رقم الطالب في برنامج الرعاية الكنسية (اختياري)
  alexandriaStudentID?: string; // رقم الطالب بالعضوية الكنسية (اختياري)
  alexandriaStudentId?: string; // alias
  alexandriaFamilyID?: string;  // رقم الأسرة بالعضوية الكنسية (اختياري)
  alexandriaFamilyId?: string;  // alias

  // Photo & Notes
  photoPath?: string;          // صورة الطالب (اختياري)
  notes?: string;              // ملاحظات إضافية (اختياري)
  deaconStatus: boolean;       // حالة شماسية
  createdAt?: string;
  updatedAt?: string;
}

export interface NIDData {
  nationalId: string;
  valid: boolean;
  birthDate: string;
  gender: string;
  governorate: string;
  age: number;
  error?: string;
}

export type ImportRowStatus = 'ready' | 'review' | 'error' | 'duplicate' | 'update';

export interface ImportIssue {
  kind: string;
  field: string;
  message: string;
}

export interface ImportRow {
  id: string;
  sheet: string;
  rowNumber: number;
  student: Student;
  status: ImportRowStatus;
  issues: ImportIssue[];
  rawGrade: string;
  gradeSuggestion: string;
  suggestionConfidence: number;
  groupKey: string;
  duplicateOf: string;
  existing?: Student;
}

export interface ImportSheet {
  name: string;
  stage: string;
  rowsFound: number;
  warning?: string;
}

export interface ImportPreview {
  sessionId: string;
  sheets: ImportSheet[];
  rows: ImportRow[];
  ready: number;
  review: number;
  errors: number;
  duplicate: number;
  new: number;
  updates: number;
}

export interface ImportSession {
  id: string;
  sourceFilename: string;
  totalRows: number;
  importedCount: number;
  initialPendingCount: number;
  pendingCount: number;
  status: string;
}

export interface PendingImportRow {
  id: string;
  sessionId: string;
  stage: string;
  issueType: 'error' | 'needs_review' | 'duplicate_in_file' | 'duplicate_in_db' | 'fuzzy_name_match';
  row: ImportRow;
  groupKey: string;
  suggestedValue: string;
  suggestionConfidence: number;
  conflictRowId: string;
  rawGrade: string;
  status: string;
}

export interface PendingImportSummary {
  sessions: ImportSession[];
  pendingCount: number;
}

export interface StudentValidation {
  valid: boolean;
  message?: string;
  student: Student;
}

export interface CommitPreviewResult {
  session: ImportSession;
  batchResult: {
    inserted: number;
    updated: number;
  };
}

export type StageType = 'حضانات (KG)' | 'ابتدائي' | 'إعدادي' | 'ثانوي' | 'جامعة';
